### PR TITLE
Update URL instead of duplicating request on MultipleChoices status code

### LINKED ISSUE #10: Request body is not passed to the next request when status code is 300 Multiple Choices
When VSD sends back a status code equals to 300 Multiple Choices, the body of the original request is not passed to the next request (with responseChoice=1 appended).

The issue comes from this line https://github.com/nuagenetworks/go-bambou/blob/e56bbad500a86dec68d009d7ed3dfa71f1038c34/bambou/session.go#L234 where the body is passed using request.Body (and not by using `request.GetBody` function). In the net/http library, the `Body` attribute within the `Request` structure only gets a non empty value only if the `newRequest` function is provided with an `io.ReadCloser ` as the `Body` parameter. In our case we use a `bytes.Buffer`. 

A better way of handling this would have been to modify the value of the `URL` attribute within the original `Request` object.

### PR DESCRIPTION
**Pull request by rvichery**
Add unit test to check body passed to second request
Fixes #10

### CODE REVIEW

**Comment by rvichery** [comment]
@pdellaert can you please review the changes

File: bambou/session_test.go [hunk 1]
```diff
@@ -1213,6 +1215,61 @@ func TestSession_Send(t *testing.T) {
...
+
+			choiceMade := false
+			sameRequestBody := false
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				
+				if choiceMade {
+					w.WriteHeader(http.StatusOK)
+					var requestBody Test
+					_ = json.NewDecoder(r.Body).Decode(&requestBody)
+					if requestBody.Test == "test" {
```
Comment (pdellaert) [inline bambou/session_test.go:1240]: Shouldn't you also test that `responseChoice` is set to `1`?

**Review by pdellaert — CHANGES_REQUESTED** [review]
Think the test can be expanded/improved a little.

Also curious to understand why the code didn't work in the first place... it is using the same request, and the body from that request. So why does the body remain the same in your code, but not in the original...

File: bambou/session_test.go (hunk 1 above)
Comment (rvichery) [reply bambou/session_test.go:1240]: Yes. I added this in 0c14482

**Review by pdellaert — APPROVED** [review]

**Pull request merged**